python/GafferImageUI/LookTransformUI.py

```diff
@@ -0,0 +1,92 @@
+#  met:
+#
+#	  * Redistributions of source code must retain the above
+#		copyright notice, this list of conditions and the following
+#		disclaimer.
+#
+#	  * Redistributions in binary form must reproduce the above
+#		copyright notice, this list of conditions and the following
+#		disclaimer in the documentation and/or other materials provided with
+#		the distribution.
+#
+#	  * Neither the name of John Haddon nor the names of
+#		any other contributors to this software may be used to endorse or
+#		promote products derived from this software without specific prior
+#		written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+import Gaffer
+import GafferImage
+import IECore
+
+Gaffer.Metadata.registerNode(
+
+	GafferImage.LookTransform,
+
+	"description",
+	"""
+	OpenColorIO LookTransform\n"
+	A 'look' is a named color transform, intended to modify the look of an
+	image in a 'creative' manner (as opposed to a colorspace definion which
+	tends to be technically/mathematically defined).\n
+	Examples of looks may be a neutral grade, to be applied to film scans
+	prior to VFX work, or a per-shot DI grade decided on by the director,
+	to be applied just before the viewing transform.\n
+	OCIOLooks must be predefined in the OpenColorIO configuration before usage,
+	often reference per-shot/sequence LUTs/CCs and are applied in scene linear colorspace.\n
+	See the look plug for further syntax details.\n
+	See opencolorio.org for look configuration customization examples.
+	""",
+
+	plugs={
+
+		"look": [
+
+			"description",
+			"""
+			Look Syntax:\n
+			Multiple looks are combined with commas: 'neutral, primary'\n
+			Direction is specified with +/- prefixes: '+neutral, -primary'\n
+			Missing look 'fallbacks' specified with |: 'neutral, -primary | -primary'
+			""",
+
+			"layout:index", 0,
+
+		],
+
+		"direction": [
+
+			"description", "Specify the look transform direction",
+
+			"presetNames", IECore.StringVectorData(["Forward", "Inverse"]),
+
+			"presetValues", IECore.IntVectorData(
+				[GafferImage.LookTransform.Direction.Forward, GafferImage.LookTransform.Direction.Inverse]),
```

Comment:
`presetNames/presetValues` are really only intended for dynamic presets that are computed on the fly, and where the number of them might change. For simple fixed presets like these I think it's preferable to just use `"preset:Forward" : Direction.Forward` etc

Reply:
switched to the simple preset formatting